rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,620 @@
+it. The DCP achieves this by never overwriting a (meta)data file but instead
+treating an update to such a file as the addition of a new version of that
+file. Instead of physically removing a file version, the DCP hides that
+file version behind a deletion marker.
+
+Under certain circumstances however, the **physical deletion** of a file version
+is required. The first section of this specification describes the
+circumstances and mechanisms for logical and physical deletion of files and the
+bundles containing them. The second section details the changes to the API required to manage the deletion process.
+The final section details the process of deleting data from the DSS, and design details.
+
+### User Stories
+
+* As a user of the DSS, I would like to know what bundles have been deleted, so I can keep my personal index up-to-date. 
+* As a data wrangler of the DSS, I need to remove data from the DSS, because there was no consent from the donor to share the data.
+* As a user of the DSS, I would like the API to return a status indicating bundle deletion, so that I know what happened to a bundle that I used previously.
+
+
+### Terminology
+
+**Tombstone** - a marker placed in the DSS to indicate data that previously existed has been removed.
+**Logical Deletion** - data is made unavailable to users through the API.
+**Physical Deletion** - data is removed from the DSS storage system.
+
+## Reasons
+
+Possible reasons for deletion of (meta)data are
+
+* consent to share (meta)data derived from a donor's specimens is withdrawn by
+  that donor after the (meta)data was submitted to DCP
+  
+* that consent was never given but this fact is revealed only after the
+  (meta)data was submitted to the DCP
+  
+* storing or processing the (meta)data causes a serious disruption of service
+  within the DCP
+
+* the (meta)data was submitted to the DCP by a malicious party in violation of
+  the DCP's data sharing agreement. This includes (meta)data that violates the
+  laws or regulations in a jurisdiction applicable to the DCP or the (meta)data
+  it stores
+
+This is not an exhaustive list, and more reason maybe discovered later.
+ 
+Whether any of the reasons require a physical or logical deletion is at the
+discretion of the administrator performing the deletion. That discretion is
+guided by a standard operating procedure (SOP).
+
+## API Changes
+
+The following are changes proposed to the DSS API to facility the deletion and restoration operations of data from the 
+DCP.
+
+### File Deletion API
+
+This is a new addition to the DSS API.
+
+Only **physical deletes** exist for the file deletion API. The file deletion API is a two part process. The first 
+request will return the bundles that will be **logically deleted**
+as a side effect and a confirmation code. The second request must include the confirmation code in order to begin the
+deletion process. The request follows this swagger document:
+
+```yaml
+      /file/{uuid}:
```

Comment:
I understand why you would want to be precise on the specification of the API, though I'm not sure the complete swagger yaml is required to do this... There's a lot of weedy detail here.

Perhaps the query path, short list of parameters, and a simple table of responses would get the point across.